pagopa/io-app
# chore(Messaggi a valore legale): [IAMVL-68] Preview MVL attachments on IOS

## Short description
We implement the flow to let the user preview a PDF attachment in-app. It only supports Android at the moment and **is behind feature-flag**.

## List of changes proposed in this pull request
- add [react-native-pdf](https://www.npmjs.com/package/react-native-pdf)
- implement [a new component](https://github.com/pagopa/io-app/pull/3774/files#diff-08db2f5b09754859765cbf0bf32242d6eb0c46d0ca027324abc02b7939ca9fe2R66) to present the PDF in a modal (no test for now as we are still experimenting with this feature and might be dropped)
- update [handleDownloadResult](https://github.com/pagopa/io-app/pull/3774/files#diff-03e64fd8c3f3946d74994448b3bfe5488063b5072f85056cc9c91601e0c6deb4R52) to inject the new behavior

## How to test

1. configure `io-dev-server` to return MVLs by adding `messages: { legalCount: 2 }` in [config/config.json](https://github.com/pagopa/io-dev-api-server/blob/master/config/config.json)
2. enable both `PAGINATED_MESSAGES` and `MVL_ENABLED` [feature flags](https://github.com/pagopa/io-app/blob/master/.env.local#L57) in your `.env` file
3. run on IOS
4. open an MVL
5. tap on the PDF attachment
6. notice how it opens the preview
7. try with an image
8. notice how it opens the native menu


https://user-images.githubusercontent.com/2094604/154714381-637694bc-adb9-4fa9-964e-d5301bd6bd67.mov
